### ISSUE TITLE
Update RestPutMappingAction for deprecation warning for inculde_types

### ISSUE THREAD

**Issue by elinamaas** [elastic/elasticsearch#58675]
We receive the deprecation warning "Specifying types in put mapping requests is deprecated. To be compatible with 7.0, the mapping definition should not be nested under the type name, and the parameter include_type_name must be provided and set to false.".  

There is a corresponding [issue](https://github.com/elastic/elasticsearch/pull/38825/). But in one file it is still the same `RestPutMappingAction.java`:

```
if (includeTypeName) {
            deprecationLogger.deprecatedAndMaybeLog("put_mapping_with_types", TYPES_DEPRECATION_MESSAGE);
        } else if (type != null || isMappingSourceTyped(MapperService.SINGLE_MAPPING_NAME, sourceAsMap)) {
            throw new IllegalArgumentException("Types cannot be provided in put mapping requests, unless " +
                "the include_type_name parameter is set to true.");
        }
```

and it is should be changed to 

```
if (request.hasParam(INCLUDE_TYPE_NAME_PARAMETER) == false) {
             deprecationLogger.deprecatedAndMaybeLog("put_mapping_with_types", TYPES_DEPRECATION_MESSAGE);
        }
```

Elasticsearch 6.8.10
Java 11
Springboot 2.2.7

**Comment by jtibshirani** [comment]
Thanks @elinamaas for reporting this! In the PR you linked (#38825), we switched from issuing a warning when `include_type_name` was missing or `true`, to only issuing a warning when `include_type_name` was missing. So I think the right fix is for us to update the message to match the other REST APIs, but keep the original `if (includeTypeName) ...` check:

```
    static final String TYPES_DEPRECATION_MESSAGE = "[types removal] The parameter include_type_name " +
        "should be explicitly specified in create index requests to prepare for 7.0. In 7.0 include_type_name " +
        "will default to 'false', and requests are expected to omit the type name in mapping definitions.";
```

I'll open a PR against 6.8 to fix this.

**Comment by jtibshirani** [comment]
Thanks @elinamaas for reporting this! It looks like we missed the 'put mapping' action in the PR you linked (#38825). I'll open a PR against 6.8 to fix this.

**Comment by jtibshirani** [comment]
I just merged #38825 to address the issue. It will be available in an upcoming 6.8 patch release.
